src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -48,7 +62,16 @@ public String findElement(List<String> elements, String element) {
      * NoSuchElementException</p>
      **/
     public Double averageSumOdd(List<Integer> numbers) {
-        return null;
+
+        IntStream.range(0,numbers.size())
+                .filter(i -> i % 2 != 0)
+                .forEach(index -> numbers.set(index, numbers.get(index) - 1));
+
```

Comment:
try to complete this task with single stream

Reply:
done